cli/js/files.ts

```diff
@@ -67,8 +69,15 @@ export class File
     Closer {
   constructor(readonly rid: number) {}
 
-  write(p: Uint8Array): Promise<number> {
-    return write(this.rid, p);
+  async write(p: Uint8Array): Promise<number> {
+    let n = 0;
+    while (n < p.length) {
+      // Cannot write over MAX_BUFFER_SIZE
+      // We need to chunk the buffer
+      const buf = p.slice(n, MAX_BUFFER_SIZE + n);
+      n += await write(this.rid, buf);
+    }
+    return n;
```

Comment:
`write` is a low level function that is not supposed to do any bufferring. It might write less bytes than the length of provided buffer and it is callers responsibility to handle that. If you need buffered write try `Deno.writeAll()`.

Reply:
`Deno.writeAll()` does not return the number of bytes written, I can modify the signature and make it return it, and use `writeAll` instead of `Writer.write` in `copy`